test/extended/marketplace/marketplace_multi.go

```diff
@@ -0,0 +1,163 @@
+
+		//wait for the opsrc is created finished
+		err = wait.Poll(5*time.Second, resourceWait, func() (bool, error) {
+			output, err := oc.AsAdmin().Run("get").Args("operatorsource", "nested", "-o=jsonpath={.status.currentPhase.phase.message}", "-n", marketplaceNs).Output()
+			if err != nil {
+				e2e.Failf("Failed to create nested, error:%v", err)
+				return false, err
+			}
+			if strings.Contains(output, "has been successfully reconciled") {
+				return true, nil
+			}
+			return false, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		podNameNest, _ := oc.AsAdmin().WithoutNamespace().Run("get").Args("pods", "-n", marketplaceNs, "-l marketplace.operatorSource=nested", "-oname").Output()
+		//wait for pod is running
+		err = wait.Poll(5*time.Second, resourceWait, func() (bool, error) {
+			output, err := oc.AsAdmin().Run("get").Args(podNameNest, "-n", marketplaceNs).Output()
+			if err != nil {
+				e2e.Failf("Failed to get the pods podNameNest status, error:%v", err)
+				return false, err
+			}
+			if strings.Contains(output, "1/1") {
+				return true, nil
+			}
+			return false, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		logOfNest, _ := oc.AsAdmin().WithoutNamespace().Run("logs").Args(podNameNest, "-n", marketplaceNs).Output()
+		o.Expect(logOfNest).Should(o.ContainSubstring("decoded 0 flattened and 1 nested"))
+
+		// OCP-25675 create the opsrc that has the flatted & nested operator and one operator has error
+		opsrcYaml2, err := oc.AsAdmin().Run("process").Args("--ignore-unknown-parameters=true", "-f", opsrcYamltem, "-p", "NAME=multiformat", "NAMESPACE=kaka", "LABEL=multiformat", "DISPLAYNAME=multiformat", "PUBLISHER=multiformat", fmt.Sprintf("MARKETPLACE=%s", marketplaceNs)).OutputToFile("config.json")
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = createResources(oc, opsrcYaml2)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		//wait for the opsrc is created finished
+		err = wait.Poll(5*time.Second, resourceWait, func() (bool, error) {
+			output, err := oc.AsAdmin().Run("get").Args("operatorsource", "multiformat", "-o=jsonpath={.status.currentPhase.phase.message}", "-n", marketplaceNs).Output()
+			if err != nil {
+				e2e.Failf("Failed to create multiformat, error:%v", err)
+				return false, err
+			}
+			if strings.Contains(output, "has been successfully reconciled") {
+				return true, nil
+			}
+			return false, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		podNameMult, _ := oc.AsAdmin().WithoutNamespace().Run("get").Args("pods", "-n", marketplaceNs, "-l marketplace.operatorSource=multiformat", "-oname").Output()
+		//wait for pod is running
+		err = wait.Poll(5*time.Second, resourceWait, func() (bool, error) {
+			output, err := oc.AsAdmin().Run("get").Args(podNameMult, "-n", marketplaceNs).Output()
+			if err != nil {
+				e2e.Failf("Failed to get the pods podNameMult status, error:%v", err)
+				return false, err
+			}
+			if strings.Contains(output, "1/1") {
```

Comment:
Is this sufficient to determine pod readiness? We should probably check the [`Ready` status condition](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions) instead.

Reply:
Hmm seems like the "Ready status consition" uneasy to get from "oc get" command. Will search about it. Thank you for your all comment @njhale .